src/reader/parser.rs

```diff
@@ -100,6 +100,10 @@ impl Parser {
                         return Err(Error::UnexpectedToken("--".to_string()));
                     }
                 }
+                if len < 5 {
+                    // We do not have any text content
+                    return Err(Error::UnexpectedToken("<!-->".to_string()));
```

Comment:
I think, it would be better to write something more descriptive, because this path will be taken also for the `<!--->` input and having `<!-->` in the error could be misleading. Maybe even introduce a new error variant specially for that case (`NotClosedComment`, for example)

Reply:
I have changed approach. I am not creating new error with the new approach. The new error message is not great but I believe it is for an other PR and not this one. Would you be ok with that?

Reply:
Yes, that is OK. We need to rework our errors anyway